### PR TITLE
AD FS Validation + Tests (Squashed)

### PR DESCRIPTION
**Pull request by iambmelt**
This PR contains the changes implemented in #752 with a flattened commit history.

### CODE REVIEW

File: adal/src/main/java/com/microsoft/aad/adal/ADALError.java [hunk 1]
```diff
@@ -314,7 +314,7 @@
      * Failed to use AndroidKeyStore.
      */
     ANDROIDKEYSTORE_FAILED("Failed to use AndroidKeyStore"),
-
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/ADALError.java]: nit: cleanup all the extra lines in this file

File: adal/src/main/java/com/microsoft/aad/adal/ADFSWebFingerValidator.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    /**
+     * Constant identifying trust between two realms.
+     */
+    private static final URI TRUSTED_REALM_REL;
+
+    static {
+        try {
+            TRUSTED_REALM_REL = new URI("http://schemas.microsoft.com/rel/trusted-realm");
+        } catch (URISyntaxException e) {
+            // will not throw
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/ADFSWebFingerValidator.java]: >// will not throw [](start = 12, length = 17)

nit: fix this comment?

File: adal/src/main/java/com/microsoft/aad/adal/ADALError.java (hunk 1 above)
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/ADALError.java]: This _is_ cleanup. There are four whitespace characters preceding the carriage return on each of these lines.

screen shot 2017-01-11 at 10 08 26 am

Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/ADALError.java]: Per offline discussion, will reintroduce so as to ease downstream merging.

Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/ADALError.java]: Resolved.

File: adal/src/main/java/com/microsoft/aad/adal/ADFSWebFingerValidator.java (hunk 2 above)
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/ADFSWebFingerValidator.java]: Resolved.

**Comment by addev-ashish** [comment]
:shipit:

**Comment by iambmelt** [comment]
@addev-ashish if https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/778 looks good, I'll merge it into this PR prior to mainlining

**Comment by iambmelt** [comment]
@addev-ashish I've merged w/ `dev` -- changes should be good 👍

**Comment by addev-ashish** [comment]
:shipit:

**Pull request merged**